cli/exchange/node.go

```diff
@@ -465,3 +480,341 @@ func NodeListErrors(org string, credToUse string, node string, long bool) {
+			return nil, errors.New(errorString)
+		}
+
+		for _, serviceDef := range services {
+			return &serviceDef, nil
+		}
+
+		return nil, errors.New("should not get here")
+
+	} else {
+		// get the hightest
+		if len(services) == 0 {
+			errorString = msgPrinter.Sprintf("expecting at least 1 service for %v %v %v, but get 0 service", serviceUrl, serviceVersion, searchVersion)
+			return nil, errors.New(errorString)
+		}
+
+		vRange, _ := semanticversion.Version_Expression_Factory("0.0.0")
+		var err error
+		if serviceVersion != "" {
+			if vRange, err = semanticversion.Version_Expression_Factory(serviceVersion); err != nil {
+				errorString = msgPrinter.Sprintf("version range %v in error: %v", serviceVersion, err)
+				return nil, errors.New(errorString)
+			}
+		}
+
+		highest, serviceDef, _, err := exchange.GetHighestVersion(services, vRange)
+		if err != nil {
+			return nil, err
+		}
+
+		if highest == "" {
+			errorString = msgPrinter.Sprintf("Cannot find service given version range: %v", serviceVersion)
+			return nil, errors.New(errorString)
+		}
+		return &serviceDef, nil
+
+	}
+}
+
+func compareUserInput(sdefUserInputs []exchange.UserInput, cmdUserInputs []policy.Input) (bool, error) {
+	msgPrinter := i18n.GetMessagePrinter()
+	serviceUserInputsMap := make(map[string]exchange.UserInput)
+	var cmdUserInputNames []string
+	var serviceInputName string
+	var serviceInput exchange.UserInput
+	var errorString string
+
+	for _, u := range cmdUserInputs {
+		cmdUserInputNames = append(cmdUserInputNames, u.Name)
+	}
+
+	// if service has a userinput with no default value AND userInput is not set in input body, return err
+	for _, serviceInput = range sdefUserInputs {
+		serviceInputName = serviceInput.Name
+		serviceUserInputsMap[serviceInputName] = serviceInput
+
+		if serviceInput.DefaultValue == "" {
+			valueDefined := sliceContains(cmdUserInputNames, serviceInputName)
+			if !valueDefined {
+				errorString = msgPrinter.Sprintf("%v need to be defined in userInput", serviceInputName)
+				return false, errors.New(errorString)
+			}
+		}
+	}
+
+	// the variables in userInput.Inputs are defined in the service, if not, return true with warning message, check values are correct types
+	var policyInputName string
+	var policyInputValue interface{}
+	var ok bool
+	var inputNameNotDefinedInService []string
+
+	for _, policyInput := range cmdUserInputs {
+		policyInputName = policyInput.Name
+		policyInputValue = policyInput.Value
+
+		serviceInput, ok = serviceUserInputsMap[policyInputName]
+		if !ok {
+			// give back a warning
+			inputNameNotDefinedInService = append(inputNameNotDefinedInService, policyInputName)
+		} else {
+			if err := cutil.VerifyWorkloadVarTypes(policyInputValue, serviceInput.Type); err != nil {
+				errorString = msgPrinter.Sprintf("Error validating type in user input %v. Error: %v", policyInputName, err)
```

Comment:
Please change this error message to:
errorString = msgPrinter.Sprintf("Error validating the type of userInput variable %v, error: %v", policyInputName, err)